### PR TITLE
implement waiting for a package to be published into the repo

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: obal/data/roles/build_package/defaults/main.yaml [hunk 1]
```diff
@@ -8,3 +8,4 @@ build_package_tito_releaser_args: []
 build_package_wait: true
 build_package_download_logs: false
 build_package_download_rpms: false
+build_package_waitrepo: false
```
Comment (ehelms) [inline obal/data/roles/build_package/defaults/main.yaml]: I would honestly love if this were true by default.

File: obal/data/roles/build_package/tasks/waitrepo.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    register: build_package_koji_task_info
+
+  - name: "Filter {{ build_package_build_system }} output"
+    set_fact:
+      build_package_koji_task_data: "{{ build_package_koji_task_info.results | map(attribute='stdout') | map('from_yaml') | list }}"
+
+  - name: "Wait for {{ build_package_koji_command }} to regenerate the repo"
+    command: "{{ build_package_koji_command }} wait-repo --build={{ item['Build'] | regex_replace(' .*', '') }} {{ item['Request Parameters']['Build Target'] | regex_replace('-candidate$', '') }}-build"
+    with_items: "{{ build_package_koji_task_data }}"
+  when: build_package_build_system == 'koji' and build_package_waitrepo|bool
```
Comment (ehelms) [inline obal/data/roles/build_package/tasks/waitrepo.yml]: Can this be placed above the block? e.g.

```
- when: blahl blahb blahb
  block:
    -
```

I feel like given yaml this should work and is in my opinion much more readable than the dangling when

File: obal/data/roles/build_package/defaults/main.yaml (hunk 1 above)
Comment (evgeni) [reply obal/data/roles/build_package/defaults/main.yaml]: We can do that, the only dependency is build_wait, and that's already true by default.

File: obal/data/roles/build_package/tasks/waitrepo.yml (hunk 2 above)
Comment (evgeni) [reply obal/data/roles/build_package/tasks/waitrepo.yml]: My memory says it does not, but I'll try out to be sure tomorrow.

**Comment by evgeni** [comment]
@ehelms updated!

File: obal/data/roles/build_package/tasks/waitrepo.yml (hunk 2 above)
Comment (evgeni) [reply obal/data/roles/build_package/tasks/waitrepo.yml]: It did \o/

File: obal/data/roles/build_package/tasks/waitrepo.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+- when: build_package_build_system == 'koji' and build_package_waitrepo|bool
```
Comment (ehelms) [inline obal/data/roles/build_package/tasks/waitrepo.yml]: Nice! We should do this more often, feels more like a traditional `if-then` relationship to me.

**Pull request merged**